### ISSUE TITLE
Argument null exception in MSBuild condition

### ISSUE THREAD

**Issue by jmue** [Caphyon/clang-power-tools#158]
When I try to compile a project I get the following output:

```
Invoke-Expression : Das Argument kann nicht an den Parameter "Command" gebunden werden, da es NULL ist.In 
D:\Users\muellerj\AppData\Local\Microsoft\VisualStudio\15.0_00aec18c\Extensions\mlnt0wvb.2aa\clang-build.ps1:1066 Zeichen:36
+   [bool] $res = (Invoke-Expression $expression) -eq $true
+                                    ~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Invoke-Expression], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Microsoft.PowerShell.Commands.InvokeExpressionCommand
```

Is there some verbosity switch, so that I can check which expression is causing the issue?

**Comment by cpp-red-lion** [comment]
VS Settings:
`[TOOLS] > [Options] > [Clang Power Tools] > [General] > Verbose Mode = True`

**Comment by jmue** [comment]
Thank you, verbose output is:

```
...
AUSFšHRLICH: SET_VAR Keyword: Win32Proj
AUSFšHRLICH: SET_VAR RootNamespace: ServiceBase
AUSFšHRLICH: SET_VAR SccProjectName: Rational ClearCase
AUSFšHRLICH: SET_VAR SccLocalPath: Rational ClearCase
AUSFšHRLICH: SET_VAR SccProvider: Rational ClearCase
AUSFšHRLICH: SET_VAR SccAuxPath: Rational ClearCase
AUSFšHRLICH: SET_VAR ConfigurationType: StaticLibrary
AUSFšHRLICH: SET_VAR UseDebugLibraries: true
AUSFšHRLICH: SET_VAR CharacterSet: Unicode
AUSFšHRLICH: SET_VAR ConfigurationType: StaticLibrary
AUSFšHRLICH: SET_VAR UseDebugLibraries: true
AUSFšHRLICH: SET_VAR CharacterSet: Unicode
AUSFšHRLICH: SET_VAR ConfigurationType: StaticLibrary
AUSFšHRLICH: SET_VAR UseDebugLibraries: false
AUSFšHRLICH: SET_VAR CharacterSet: Unicode
AUSFšHRLICH: SET_VAR ConfigurationType: StaticLibrary
AUSFšHRLICH: SET_VAR UseDebugLibraries: false
AUSFšHRLICH: SET_VAR CharacterSet: Unicode
AUSFšHRLICH: 
Sanitizing <Dir>:\<PathToProject>\<ProjectName>.vcxproj
Invoke-Expression : Das Argument kann nicht an den Parameter "Command" gebunden werden, da es NULL ist.In 
D:\Users\muellerj\AppData\Local\Microsoft\VisualStudio\15.0_00aec18c\Extensions\mlnt0wvb.2aa\clang-build.ps1:1066 Zeichen:36
+   [bool] $res = (Invoke-Expression $expression) -eq $true
+                                    ~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Invoke-Expression], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Microsoft.PowerShell.Commands.InvokeExpressionCommand
```

**Comment by jmue** [comment]
Ok I found the issue, my conditions uses a 'Selector.Contains()' statement. I attached a somple project:

[clang-pt.zip](https://github.com/Caphyon/clang-power-tools/files/1551810/clang-pt.zip)

**Comment by hero101111** [comment]
Hi,

Indeed, the issue arose from the `$(Selector.Contains(Three))` expression.
`Three` here is intended to be a string, but since there no quotes the interpreter thinks it's a command or a variable.

We do not support string parameters that are not wrapped in quotes.

To get your project to compile just wrap `Three` in single quotes, as follows:
`$(MySelector.Contains('Three'))`

I've tested and it works in both Visual C++ and ClangPowerTools.

Regards,
Gabriel

**Comment by jmue** [comment]
But it does also nit compile if there is an expression within the Contains statement:

```
  <PropertyGroup>
    <MySelector>One,Two,Three</MySelector>
    <MySelectorElement>Four</MySelectorElement>
  </PropertyGroup>
```

A statement like ```Condition="$(MySelector.Contains($(MySelectorElement)))``` raises the error as ```Condition="$(MySelector.Contains('$(MySelectorElement)'))``` does.

**Comment by jmue** [comment]
There are some other condition statements like ```Condition="HasTrailingSlash($(MyPath))"``` that are causing the issue too.

```
  <PropertyGroup>
    <MyPath>C:\Tmp\</MyPath>
  </PropertyGroup>
```

**Comment by hero101111** [comment]
We're investigating these issues.
You'll hear from us when we have a solution.

**Comment by hero101111** [comment]
Hi,

I've pushed a fix for these issues.
Grab the latest `clang-build.ps1` from Github and let us know if it solves your project specific issues.

Regards,
Gabriel

---

This fix is currently live only for the github repository.
The latest Visual Studio extension (`2.2`) does not contain the fix. It will be included in the next release.

Until then, as a workaround, you can manually replace `clang-build.ps1` in the VS extension directory with the latest from github. To locate the extension directory search for `clang-build.ps1` in `%LOCALAPPDATA%`

Don't forget to restart Visual Studio after replacing the script file.

**Comment by hero101111** [comment]
We fixed these issues in version `2.3` we just released.
